Source/Core/InputCommon/ControllerInterface/ExpressionParser.cpp

```diff
@@ -427,13 +427,12 @@ class Parser
 
   ExpressionParseStatus Unary(ExpressionNode** expr_out)
   {
-    ExpressionParseStatus status;
-
     if (IsUnaryExpression(Peek().type))
     {
       Token tok = Chew();
       ExpressionNode* atom_expr;
-      if ((status = Atom(&atom_expr)) != EXPRESSION_PARSE_SUCCESS)
+      ExpressionParseStatus status = Atom(&atom_expr);
+      if (status != EXPRESSION_PARSE_SUCCESS && status != EXPRESSION_PARSE_NO_DEVICE)
```

Comment:
You should probably make this a conditional check a boolean function with an informative name, considering it's used 3 times verbatim